src/plugin/PluginManager.php

```diff
@@ -626,8 +629,29 @@ public function registerEvents(Listener $listener, Plugin $plugin) : void{
 						throw new PluginException("Event handler " . Utils::getNiceClosureName($handlerClosure) . "() declares invalid @" . ListenerMethodTags::HANDLE_CANCELLED . " value \"" . $tags[ListenerMethodTags::HANDLE_CANCELLED] . "\"");
 				}
 			}
+			$noConcurrentCall = false;
+			if(isset($tags[ListenerMethodTags::NO_CONCURRENT_CALL])){
+				if(!is_a($eventClass, AsyncEvent::class, true)){
+					throw new PluginException(sprintf(
+						"Event handler %s() declares @%s for non-async event of type %s",
+						Utils::getNiceClosureName($handlerClosure),
+						ListenerMethodTags::NO_CONCURRENT_CALL,
+						$eventClass
+					));
+				}
+				switch(strtolower($tags[ListenerMethodTags::NO_CONCURRENT_CALL])){
+					case "true":
+					case "":
+						$noConcurrentCall = true;
+						break;
+					case "false":
+						break;
+					default:
+						throw new PluginException("Event handler " . Utils::getNiceClosureName($handlerClosure) . "() declares invalid @" . ListenerMethodTags::NO_CONCURRENT_CALL . " value \"" . $tags[ListenerMethodTags::NO_CONCURRENT_CALL] . "\"");
+				}
```

Comment:
Not related to this PR, but both the API and this handler would be much nicer if we use attributes instead:

```php
#[\pocketmine\event\AsyncHandler]
public function handler(XxxEvent $event) : Promise { ... }

#[\pocketmine\event\AsyncHandler(concurrent: true)]
public function handler(XxxEvent $event) : Promise { ... }
```